nomad/state/state_store.go

```diff
@@ -303,6 +303,8 @@ func (s *StateStore) UpsertPlanResults(index uint64, results *structs.ApplyPlanR
 
 	// Upsert followup evals for allocs that were preempted
 	for _, eval := range results.PreemptionEvals {
```

Comment:
we should not touch eval timestamps here because then the same eval id will have different timestamp values in different servers. Delete this code, the callers should be setting the two new fields instead.